cmake/modules/RustExt.cmake

```
@@ -0,0 +1,26 @@
+if(USE_RUST_EXT AND NOT USE_RUST_EXT EQUAL OFF)
```

We really need to stop overloading cmake bool and string. cmake is a simple thing and is easily confused. In the meantime us this instead. OFF is only one of the many valid values for cmake bool false.
```suggestion
if(USE_RUST_EXT AND NOT ${USE_RUST_EXT} MATCHES ${IS_FALSE_PATTERN})
```

---

I am not a great CMake user so just let me know what the best way to do things is. I just have no clue what the "right way" in CMake is.

---

I think `if(USE_RUST_EXT)` should be fine.

---

_if(<constant>)
True if the constant is 1, ON, YES, TRUE, Y, or a non-zero number. False if the constant is 0, OFF, NO, FALSE, N, IGNORE, NOTFOUND, the empty string, or ends in the suffix -NOTFOUND. Named boolean constants are case-insensitive. If the argument is not one of these constants, it is treated as a variable._